### ISSUE TITLE
JsonArray of type with custom deserializer not properly deserialized.

### ISSUE THREAD

**Issue by etay2000** [eclipse/yasson#67]
```
public enum FooEnum {
        FOO,
        BAR,
        BAZ;
    }

// Custom Deserializer registered on JsonbConfig
public class FooEnumDeserializer implements JsonbDeserializer<FooEnum> {

    @Override
    public FooEnum deserialize(JsonParser jsonParser, DeserializationContext deserializationContext, Type type) {
        String name = jsonParser.getString().toUpperCase();
        return FooEnum.valueOf(name);
    }

}

Type type = new HashSet<FooEnum>() {}.getClass().getGenericSuperclass();
JsonArray arr = Json.createArrayBuilder()
                .add("foo")
                .add("bar")
                .add("baz")
                .build();

Set<FooEnum> values = getJsonb().fromJson(arr.toString(), type);
// values only contains first item from the JsonArray [ 'foo' ]

```

Only the first item in the array is deserialized, so the resulting collection always only contains one item.

**Comment by Verdent** [comment]
I was not able to reproduce this issue against master branch. Could you please tell me which version do you use it against?

**Comment by etay2000** [comment]
I just double checked and I can reproduce it on 1.0.1. I still only get 1 value in the resulting set. 
The last line in my example should actually be: 
`// values only contains first item from the JsonArray [ 'FOO' ]`
)

**Comment by m0mus** [comment]
There is a chance that this issue is fixed in master but not fixed in 1.0.1. 
@Verdent, can you please check it. 
If so, close the issue as fixed.

**Comment by Verdent** [comment]
I have verified that this is issue at 1.0.1 but no longer at 1.0.2. I would say it is ok to close this issue.
